tests/migrations/test_update_legacy_filtered_positives.py

```diff
@@ -107,92 +116,168 @@ def test_update_legacy_filtered_positives_exception_raised_if_user_enters_invali
+    update_legacy_filtered_positives.run("crawler.config.integration", start_datetime, end_datetime)
+
+    # ensure no database connections/updates are made
+    mock_update_mongo.assert_not_called()
+    mock_update_mlwh.assert_not_called()
+
+
+def test_update_legacy_filtered_positives_returns_early_invalid_end_datetime(config, mock_helper_database_updates):
+    mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+    start_datetime = "201016_1600"
+    end_datetime = "not a real datetime"
+
+    mock_valid_datetime_string.side_effect = [True, False]
+    update_legacy_filtered_positives.run("crawler.config.integration", start_datetime, end_datetime)
+
+    # ensure no database connections/updates are made
+    mock_update_mongo.assert_not_called()
+    mock_update_mlwh.assert_not_called()
+
+
+def test_update_legacy_filtered_positives_returns_early_datetime_post_fields_set_date(
+    config, mock_helper_database_updates
+):
+    mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+    start_datetime = "201016_1600"
+    end_datetime = "201217_1600"
+
+    mock_valid_datetime_string.return_value = True
+    update_legacy_filtered_positives.run("crawler.config.integration", start_datetime, end_datetime)
+
+    # ensure no database connections/updates are made
+    mock_update_mongo.assert_not_called()
+    mock_update_mlwh.assert_not_called()
+
+
+def test_update_legacy_filtered_positives_returns_early_start_datetime_after_end_datetime(
+    config, mock_helper_database_updates
+):
+    mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+    start_datetime = "201017_1200"
+    end_datetime = "201016_1600"
+    update_legacy_filtered_positives.run("crawler.config.integration", start_datetime, end_datetime)
 
+    # ensure no database connections/updates are made
     mock_update_mongo.assert_not_called()
     mock_update_mlwh.assert_not_called()
 
 
 def test_update_legacy_filtered_positives_catches_error_connecting_to_mongo(
-    mock_helper_database_updates, mock_v0_version_set
+    mock_helper_database_updates, mock_filtered_positive_fields_set
 ):
+    mock_filtered_positive_fields_set.return_value = False
+
+    mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
+    mock_update_mongo.side_effect = Exception("Boom!")
+
     with pytest.raises(Exception):
-        with patch(
-            "migrations.update_legacy_filtered_positives.legacy_mongo_samples",
-            side_effect=Exception("Boom!"),
-        ):
-            mock_v0_version_set.return_value = False
-            mock_update_mongo, mock_update_mlwh = mock_helper_database_updates
-            update_legacy_filtered_positives.run("crawler.config.integration")
+        update_legacy_filtered_positives.run("crawler.config.integration", start_date_input, end_date_input)
 
-            mock_update_mongo.assert_not_called()
-            mock_update_mlwh.assert_not_called()
+    mock_update_mongo.assert_called_once()
```

Comment:
I'm not entirely sure anything gets called after an exception is caught in the `with pytest.raises(Exception):` block. Would you be able to check this (by ensuring the test fails if you assert a call was made)?

Reply:
Looks like I'm wrong on this one: inverting the check causes the tests to fail, indicating that the assertions are called